zulipterminal/model.py

```diff
@@ -232,12 +232,18 @@ def toggle_message_star_status(self, message: Dict[str, Any]) -> None:
     def mark_message_ids_as_read(self, id_list: List[int]) -> None:
         if not id_list:
             return
-        self.client.update_message_flags({
+        response = self.client.update_message_flags({
             'messages': id_list,
             'flag': 'read',
             'op': 'add',
         })
-        set_count(id_list, self.controller, -1)  # FIXME Update?
+        # We should only update the local cache of counts if the
+        # message is marked successfully this way; server events
+        # will trigger updates of the index for other cases.
+        if response['result'] == 'success':
+            # FIXME: Failure should check partial success?
+            # Maybe we should update each message flag individually?
```

Comment:
Updating each message flag individually may result in a large number of api requests to the server and hence the server may block the client from sending further requests.

Reply:
I wasn't necessarily suggesting that approach, but I'm happy to drop this commit for now until we agree on a solution.

How do you like the other changes? Good to merge?

Reply:
yeah other changes look nice to me.

Reply:
OK, have dropped this commit - will merge now.